.github/workflows/default.yaml

```diff
@@ -122,7 +130,11 @@ jobs:
 
       - name: Run build on Linux
         if: runner.os == 'Linux'
-        run: ./test-builds.sh
+        env:
+          CC: ${{ matrix.compiler.CC }}
+          CXX: ${{ matrix.compiler.CXX }}
```

Comment:
Is it too dangerous to make this a job-specific environment (rather than step-specific environment) because some earlier steps may pick up a not-yet-installed compiler?

If it is too dangerous, we would need to duplicate this for MacOS. The latter is best done now (rather than when we add another MacOS compiler variation) IMO but I do not insist on that.

Reply:
[Seems to be working](https://github.com/squid-cache/squid/actions/runs/9049941451). It might come back and haunt us in the future (e.g. if we need to specify a path to run a build in some bizarre environment) but for now it seems to be working